sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleClient.scala

```diff
@@ -450,13 +452,22 @@ class RapidsShuffleClient(
 
           val stats = tx.getStats
 
+          val toDelete = new ArrayBuffer[ShuffleReceivedBufferId]()
+
           // hand buffer off to the catalog
           buffMetas.foreach { consumed: ConsumedBatchFromBounceBuffer =>
             val bId = track(consumed.contigBuffer, consumed.meta)
-            consumed.handler.batchReceived(bId.asInstanceOf[ShuffleReceivedBufferId])
+            if (!consumed.handler.batchReceived(bId)) {
+              toDelete.append(bId)
```

Comment:
Why collect them instead of deleting them as they are handled?  Seems like we can ditch the `ArrayBuffer` completely and track the number of times we threw away in a local variable, performing a debug log if that variable is non-zero at the end.

Reply:
will do, that's better!